app/lib/actions/katello/capsule_content/sync_capsule.rb

```diff
@@ -3,24 +3,28 @@ module Katello
     module CapsuleContent
       class SyncCapsule < ::Actions::EntryAction
         include Actions::Katello::PulpSelector
+        # rubocop:disable Metrics/MethodLength
         def plan(smart_proxy, options = {})
           action_subject(smart_proxy)
           environment = options[:environment]
           content_view = options[:content_view]
           repository = options[:repository]
           skip_metadata_check = options.fetch(:skip_metadata_check, false)
-
+          smart_proxy_history_ids = []
           sequence do
             repos = repos_to_sync(smart_proxy, environment, content_view, repository)
 
             repos.in_groups_of(Setting[:foreman_proxy_content_batch_size], false) do |repo_batch|
               concurrence do
                 repo_batch.each do |repo|
-                  plan_pulp_action([Actions::Pulp::Orchestration::Repository::SmartProxySync,
-                                    Actions::Pulp3::CapsuleContent::Sync],
+                  action_needed = action_needed?(repo, smart_proxy, skip_metadata_check)
```

Comment:
instead of doing this on a per-repo basis, could we make this a scope on the Repository object?    It might have te in the repos_to_sync() method, but being able to build that into the list would be a lot more efficient